libcloud/common/dimensiondata.py

```diff
@@ -217,6 +218,28 @@ def get_resource_path_api_2(self):
         return ("%s/%s/%s" % (self.api_path_version_2, self.api_version_2,
                               self._get_orgId()))
 
+    def wait_for_state(self, state, func, *args, **kwargs):
+        """
+        Wait for the function which returns a instance
+        with field status to match
+
+        Keep polling func until one of the desired states is matched
+
+        :param state: Either the desired state (`str`) or a `list` of states
+        :type  state: ``str`` or ``list``
+
+        :param  func: The function to call, e.g. ex_get_vlan
+        :type   func: ``function``
+
+        :param  kwargs: The arguments for func
+        :type   kwargs: Keyword arguments
+        """
+        while(True):
```

Comment:
Might also be a good idea to add some kind of timeout functionality?

Since now it could potentially block forever...